### PR TITLE
Added retries for Vertex LLM

### PR DESCRIPTION
**Pull request by lkuligin**
#7217

### CODE REVIEW

File: langchain/llms/base.py [hunk 1]
```diff
@@ -34,6 +55,38 @@ def _get_verbosity() -> bool:
...
+    # Wait 2^x * 1 second between each retry starting with
+    # 4 seconds, then up to 10 seconds, then 10 seconds afterwards
+    if len(error_types) == 0:
+        return lambda x: x
+    elif len(error_types) == 1:
+        retries = retry_if_exception_type(error_types[0])
+    else:
+        init = retry_if_exception_type(error_types[0])
+        retries = functools.reduce(
+            lambda x, y: x | retry_if_exception_type(y),  # type: ignore
```
Comment (hwchase17) [inline langchain/llms/base.py:77]: is there a simpler way to do this?

File: langchain/llms/base.py [hunk 2]
```diff
@@ -34,6 +55,38 @@ def _get_verbosity() -> bool:
     return langchain.verbose
 
 
+def create_base_retry_decorator(
+    error_types: List[Type[BaseException]], max_retries: int = 1
+) -> Callable[[Any], Any]:
+    """Create a retry decorator for a given LLM and provided list of error types."""
+    import logging
```
Comment (hwchase17) [inline langchain/llms/base.py:62]: this and the logger initialization can go at top of file i think

**Comment by hwchase17** [comment]
like the refactor out

File: langchain/llms/base.py (hunk 2 above)
Comment (lkuligin) [reply langchain/llms/base.py]: Done

File: langchain/llms/base.py (hunk 1 above)
Comment (lkuligin) [reply langchain/llms/base.py]: Please, take a look, maybe it's more readable without `functools`.

**Review by hwchase17 — APPROVED** [review]
nice - thanks!

**Pull request merged**